manim/mobject/stats.py

```diff
@@ -91,24 +91,68 @@ def __init__(self, x, y, graph_scene=None, **kwargs):
             )
 
         def function(t):
-            if (
-                t < self._hist_points[0][0]
-                or t > self._hist_points[len(self._hist_points) - 1][0]
-            ):
-                return self._coords_to_point(t, np.nan)
+            if t < self._graph_points[0][0] or t > self._graph_points[-1][0]:
+                point = (t, np.nan)
+
+            else:
+                point = self._points_search(t)
 
-            for x, y in self._hist_points:
-                if t == x:
-                    return self._coords_to_point(x, y)
+                if point is None:
+                    point = self._lines_search(t)
 
-            for line in self._lines:
-                if t > line["domain"][0] and t < line["domain"][1]:
-                    return self._coords_to_point(t, line["func"](t))
+            return self._coords_to_point(*point)
 
         super().__init__(function, **kwargs)
 
+    def _points_search_helper(self, low, high, x):
+        mid = self._graph_points[(high + low) // 2]
+        return (
+            (
+                mid
+                if mid[0] == x
+                else self._points_search_helper(low, (high + low) // 2 - 1, x)
```

Comment:
Comment: in general, it is better in python to not user recursion. But we can come back to this if it ever becomes a nuisance